File: app/src/main/java/com/example/roomapplication/MainActivity.java
package com.example.roomapplication;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Spinner;
import android.widget.Toast;

import com.example.roomapplication.roomdb.Note;
import com.example.roomapplication.roomdb.NoteDatabase;

import java.util.Objects;

public class MainActivity extends AppCompatActivity {

    public NoteDatabase noteDB;
    public EditText nameEditTxt, ageEditTxt, jobTitleEditTxt;
    public Button addBtn, getBtn;

    public Spinner genderSpinner;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        noteDB = NoteDatabase.getInstance(this);

        intiView();

        String[] genderItems = new String[]{"Male", "Female"};

        ArrayAdapter<String> genderAdapter = new ArrayAdapter<>(this, android.R.layout.simple_spinner_dropdown_item, genderItems);
        genderSpinner.setAdapter(genderAdapter);

        addBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                addData();
            }
        });
        getBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getData();

            }
        });


    }

    private void intiView() {
        nameEditTxt = findViewById(R.id.name);
        ageEditTxt = findViewById(R.id.age);
        jobTitleEditTxt = findViewById(R.id.jobTitle);
        genderSpinner = findViewById(R.id.gender);
        addBtn = findViewById(R.id.addBtn);
        getBtn = findViewById(R.id.getBtn);
    }

    private void clearInputs() {
        Objects.requireNonNull(nameEditTxt.getText()).clear();
        Objects.requireNonNull(ageEditTxt.getText()).clear();
        Objects.requireNonNull(jobTitleEditTxt.getText()).clear();
    }

    private void addData() {
        Note note = noteCreator();
        inputIsEmpty(note);
        clearInputs();
    }

    private void getData() {
        showToastMessages(getString(R.string.get_data_message));
        goToResult();
    }

    private void inputIsEmpty(Note note) {
        if (!nameEditTxt.getText().toString().matches("") &&
                !ageEditTxt.getText().toString().matches("") &&
                !jobTitleEditTxt.getText().toString().matches("") &&
                !genderSpinner.getSelectedItem().toString().matches("")) {

            noteDB.getNoteDao().insert(note);
            showToastMessages(getString(R.string.add_successfuly));
        } else {
            showToastMessages(getString(R.string.please_fill_data));
        }
    }

    private Note noteCreator() {
        return new Note(nameEditTxt.getText().toString(), ageEditTxt.getText().toString(), jobTitleEditTxt.getText().toString(), genderSpinner.getSelectedItem().toString());
    }

    private void goToResult() {
        startActivity(new Intent(MainActivity.this, ResultActivity.class));
    }

    private void showToastMessages(String input) {
        Toast.makeText(getApplicationContext(), input, Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/com/example/roomapplication/MainActivity.kt
package com.example.roomapplication

import android.content.Intent
import android.os.Bundle
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
import com.example.roomapplication.roomdb.Note
import com.example.roomapplication.roomdb.NoteDatabase
import com.example.roomapplication.roomdb.NoteDatabase.Companion.getInstance
import java.util.*

class MainActivity : AppCompatActivity() {
  lateinit  var noteDB: NoteDatabase
   lateinit var nameEditTxt: EditText
   lateinit var ageEditTxt: EditText
  lateinit  var jobTitleEditTxt: EditText
   lateinit var addBtn: Button
   lateinit var getBtn: Button
   lateinit var genderSpinner: Spinner
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        noteDB = getInstance(this)
        intiView()
        val genderItems = arrayOf(getString(R.string.male), getString(R.string.female))
        val genderAdapter = ArrayAdapter(this, android.R.layout.simple_spinner_dropdown_item, genderItems)
        genderSpinner.adapter = genderAdapter
        addBtn.setOnClickListener { addData() }
        getBtn.setOnClickListener { data }
    }

    private fun intiView() {
        nameEditTxt = findViewById(R.id.name)
        ageEditTxt = findViewById(R.id.age)
        jobTitleEditTxt = findViewById(R.id.jobTitle)
        genderSpinner = findViewById(R.id.gender)
        addBtn = findViewById(R.id.addBtn)
        getBtn = findViewById(R.id.getBtn)
    }

    private fun clearInputs() {
        Objects.requireNonNull(nameEditTxt!!.text).clear()
        Objects.requireNonNull(ageEditTxt!!.text).clear()
        Objects.requireNonNull(jobTitleEditTxt!!.text).clear()
    }

    private fun addData() {
        val note = noteCreator()
        inputIsEmpty(note)
        clearInputs()
    }

    private val data: Unit
        get() {
            showToastMessages(getString(R.string.get_data_message))
            goToResult()
        }

    private fun inputIsEmpty(note: Note) = if (nameEditTxt!!.text.toString() != "" &&
            ageEditTxt!!.text.toString() != "" &&
            jobTitleEditTxt!!.text.toString() != "" &&
            genderSpinner!!.selectedItem.toString() != "") {
        noteDB!!.noteDao?.insert(note)
        showToastMessages(getString(R.string.add_successfuly))
    } else {
        showToastMessages(getString(R.string.please_fill_data))
    }

    private fun noteCreator(): Note {
        return Note(nameEditTxt!!.text.toString(), ageEditTxt!!.text.toString(), jobTitleEditTxt!!.text.toString(), genderSpinner!!.selectedItem.toString())
    }

    private fun goToResult() {
        startActivity(Intent(this@MainActivity, ResultActivity::class.java))
    }

    private fun showToastMessages(input: String) {
        Toast.makeText(applicationContext, input, Toast.LENGTH_SHORT).show()
    }
}